### PR TITLE
refactor common cli argument handling

### PR DESCRIPTION
**Pull request by williballenthin**
addresses enhancement described in #449 

also removes old migration script.
also some minor style fixes as suggested by pylance in vscode.

### CODE REVIEW

File: capa/main.py [hunk 1]
```diff
@@ -282,6 +282,8 @@ def get_workspace(path, format, should_save=True):
         vw = get_shellcode_vw(path, arch="i386", should_save=should_save)
     elif format == "sc64":
         vw = get_shellcode_vw(path, arch="amd64", should_save=should_save)
+    else:
+        raise ValueError("unexpected format: " + format)
```
Comment (williballenthin) [inline capa/main.py:286]: better catch unrelated errors.

File: capa/main.py [hunk 2]
```diff
@@ -444,19 +446,157 @@ def collect_metadata(argv, sample_path, rules_path, format, extractor):
...
+
+    # disable vivisect-related logging, it's verbose and not relevant for capa users
+    set_vivisect_log_level(logging.CRITICAL)
+
+    # py2 doesn't know about cp65001, which is a variant of utf-8 on windows
+    # tqdm bails when trying to render the progress bar in this setup.
+    # because cp65001 is utf-8, we just map that codepage to the utf-8 codec.
+    # see #380 and: https://stackoverflow.com/a/3259271/87207
+    import codecs
+    codecs.register(lambda name: codecs.lookup("utf-8") if name == "cp65001" else None)
```
Comment (williballenthin) [inline capa/main.py:580]: not strictly the perfect place to put this, but its the best right now.

File: scripts/capa_as_library.py [hunk 3]
```diff
@@ -6,6 +6,7 @@
 import capa.main
 import capa.rules
 import capa.engine
+import capa.render
```
Comment (williballenthin) [inline scripts/capa_as_library.py:9]: unrelated style fix

File: scripts/lint.py [hunk 4]
```diff
@@ -320,7 +320,7 @@ def check_rule(self, ctx, rule):
         expected = capa.rules.Rule.from_yaml(rule.definition, use_ruamel=True).to_yaml()
 
         if actual != expected:
-            diff = difflib.ndiff(actual.splitlines(1), expected.splitlines(1))
+            diff = difflib.ndiff(actual.splitlines(1), expected.splitlines(True))
```
Comment (williballenthin) [inline scripts/lint.py:323]: unrelated recommendation from pylance

**Comment by williballenthin** [comment]
this is ready for review/merge.

**Review by mr-tz — APPROVED** [review]
much more elegant, thanks!

**Comment by mr-tz** [comment]
👍

**Pull request merged**